### PR TITLE
Update jobs/gpu display format

### PR DESCRIPTION
**Pull request by lukew3**
image

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201735133575781/1202059223026718) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by lukew3** [comment]
Current working version of gpu and jobs bar charts
image

File: app/assets/stylesheets/barchart.scss [hunk 1]
```diff
@@ -0,0 +1,46 @@
+$gap: 5px;
+$totalRows: 101;
```
Comment (johrstrom) [inline app/assets/stylesheets/barchart.scss:2]: I think the pattern we want here is to put these into a `_variables.scss` and import it in `application.scss`.

File: app/assets/stylesheets/barchart.scss [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+.barchart {
+  width: 90%;
+  height: 250px;
```
Comment (johrstrom) [inline app/assets/stylesheets/barchart.scss:6]: Shouldn't this be some `em` and/or %?  or if we need to force a value does `min-height` work (with `height`)?

File: app/views/jobs/_system_status.html.erb [hunk 3]
```diff
@@ -1,20 +1,62 @@
+<%- 
+    gpu_using = jobs.count { |job| job.status == 'running' && job.native[:gres].include?("gpu") }
+    gpu_requesting = jobs.count { |job| job.status == 'queued' && job.native[:gres].include?("gpu") }
+    gpu_using_pct = (gpu_using + gpu_requesting != 0) ? (gpu_using.to_f / (gpu_using + gpu_requesting) * 100).to_i : 0
+    gpu_requesting_pct = (gpu_using + gpu_requesting != 0) ? (gpu_requesting.to_f / (gpu_using + gpu_requesting) * 100).to_i : 0
+    jobs_running = jobs.count { |job| job.status == 'running' }
+    jobs_queued = jobs.count { |job| job.status == 'queued' }
+    jobs_running_pct = (jobs_running + jobs_queued != 0) ? (jobs_running.to_f / (jobs_running + jobs_queued) * 100).to_i : 0
+    jobs_queued_pct = (jobs_running + jobs_queued != 0) ? (jobs_queued.to_f / (jobs_running + jobs_queued) * 100).to_i : 0
```
Comment (johrstrom) [inline app/views/jobs/_system_status.html.erb:9]: Put all of these into `app/helpers/job_helpers.rb`. When the view renders, it'll behave just like it does now, the definitions are just somewhere else.

File: app/views/jobs/_system_status.html.erb [hunk 4]
```diff
@@ -1,20 +1,53 @@
...
+                    <p class="barLabel">
+                        GPUs in use
+                    </p>
+                </div>
+                <div class="bcBar bar-<%= cluster_info[:gpu_requesting_pct] %> bg-primary">
+                    <p class="barValue">
+                        <%= cluster_info[:gpu_requesting] %>
+                    </p>
+                    <p class="barLabel">
+                        GPUs requested
```
Comment (johrstrom) [inline app/views/jobs/_system_status.html.erb:21]:
```suggestion
                        GPU jobs queued
```

When I read requested - I questioned whether it meant `queued` or `queued + running`. Looking at the code it's actually queued, so let's call it that.

Also - to be crystal clear - it's GPU jobs that are pending, not the GPUs themselves by the looks of the calculation.

File: app/views/jobs/_system_status.html.erb [hunk 5]
```diff
@@ -1,20 +1,53 @@
...
-    </table>
+    <div class="row justify-content-center">
+        <div class="col">
+            <div class="barchart">
+                <div class="bcBar bar-<%= cluster_info[:gpu_using_pct] %> bg-primary">
+                    <p class="barValue">
+                        <%= cluster_info[:gpu_using] %>
+                    </p>
+                    <p class="barLabel">
+                        GPUs in use
```
Comment (johrstrom) [inline app/views/jobs/_system_status.html.erb:13]:
```suggestion
                        Active GPU jobs
```

File: app/views/jobs/_system_status.html.erb (hunk 4 above)
Comment (johrstrom) [reply app/views/jobs/_system_status.html.erb:21]: Happy to keep reporting jobs for now and defer #25, but whatever we show it's got to be labeled appropriately.

**Comment by lukew3** [comment]
Ideally, these charts would fetch their data from ood_core. However, since the ood_core gem isn't up-to-date and doesn't contain cluster_info, I chose to continue displaying jobs using gpu instead of actual gpu usage. Once the next version is released, I plan on correctly displaying the number of gpus in-use and queued.

File: app/assets/stylesheets/barchart.scss [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  height: 12em;
+  margin: 0 auto;
+  display: flex;
+  padding: $gap;
+  box-sizing: border-box;
+  align-items: flex-end;
+  margin-bottom: 1em;
+}
+
+.bcBar {
```
Comment (johrstrom) [inline app/assets/stylesheets/barchart.scss:12]:
```suggestion
.bc-bar {
```

This prompted me to update the style guide for OnDemand.

File: app/assets/stylesheets/_variables.scss [hunk 7]
```diff
@@ -0,0 +1,2 @@
+$gap: 5px;
+$totalRows: 101;
```
Comment (johrstrom) [inline app/assets/stylesheets/_variables.scss:2]:
```suggestion
$total-rows: 101;
```

Apparently hyphenated names are a CSS convention even if other types are allowed.

**Review by johrstrom — APPROVED** [review]
I'm ready to pull this in if you are.

**Pull request merged**